### ISSUE TITLE
Placeholder Article not updating - Alinbie#5002

### ISSUE THREAD

**Issue by Sockemseals** [worldanvil/worldanvil-bug-tracker#1261]
**World Anvil Username**: Sockemseal90

**World / Campaign / RPG Character / Manuscript Title**: Toril - Encyclopedia

**Feature**: Placeholder Article Link

**Related URLS** (To public & visible pages add screenshots otherwise): Issue occurs on all pages & worlds of mine.

**Describe the Issue**
Within a parent article I added a placeholder (Ex. @Amphail(Settlement):) then when I go to the view world side and click that link it creates a new article and opens it to edit. I then work on that article, link it to the parent article and publish. However, when I go to the parent article it remains a placeholder and does not link to the newly created article. If I click that article again, it will repeat the process and open a new article to be edited and published. 

**To Reproduce**
It always does this, even on my other world.
1) Add Placeholder within a parent article
2) Go to View world, and parent article
3) click that placeholder link
4) edit the article, don't do anything, publish, or leave as draft.
5) go back to the parent article
6) click that placeholder link
7) repeat at step 4

**Expected behavior**
Parent article should be updated with a hyperlink to the newly created article in place of the placeholder link. When going into the edit of the parent article it should say: @[Amphail](Settlement:xxxxxxxxxxxxxxxID) instead of remaining as @Amphail(Settlement). And clicking the link within the parent should take me to the new article.

**Screenshots**
If applicable, add screenshots to help explain your problem. You can add screenshots by drag & drop into this window!
![image](https://user-images.githubusercontent.com/113307379/189543357-3e434bba-9c2c-40ed-bd16-945c960891f7.png)
![image](https://user-images.githubusercontent.com/113307379/189543490-fba2654e-16e9-4595-8d28-4514648e1b83.png)
![image](https://user-images.githubusercontent.com/113307379/189543887-68eae965-8094-4cfb-9d14-af390f27538d.png)

In final Screenshot I did this process for Daggerford, then SoulLink joined as an editor and it seemed to work properly, but neither of us edited that article during that time. I tried again for Julkoun and it didn't update like Daggerford (SoulLink still an editor) but the color in the quick edit is orange, not red like the other placeholders.

**Comment by Kordishal** [comment]
Potentially same issue as #941

**Comment by iamromeo** [comment]
Hello @Sockemseals  can you try moving the colon away from the placeholder ? I am not certain this is the issue but I cannot think of anything else 

not related to #941

**Comment by Kordishal** [comment]
@iamromeo I created an article and did some tests with just basic template and articles as an editor on their world. I ran into the same issue. It does seem to update them sometime, but not consistently. 

What is also weird that it appears updated on the view side as it always links to the new article and does not create a new article. However, when I den go to edit that article it still displays the placeholder mention and not the proper linked mention.

**Comment by iamromeo** [comment]
@Kordishal can this be an issue with the article still being a draft ?

**Comment by Kordishal** [comment]
@iamromeo No that is not the issue. But I found a way to reliably have it become updated. When the article is saved again with any change not directly affecting the placeholder link then it updates the content displayed to reflect the link. It appears to me that the issue is only present on the users side and not actually in the server. 

And I was able to reproduce this issue in my world like this. Basically the article is updated, but for some reason the client only picks this up when there is some kind of save. Only tested it in Euclid.

That would also explain why when you view the article the link is correct after creating the placeholder article. Even after reloading and saving the unchanged placeholder link.
